test/unit/app/controllers/transactions/pending-tx-test.js

```diff
@@ -107,6 +106,36 @@ describe('PendingTransactionTracker', function () {
       pendingTxTracker._checkPendingTx(txMetaNoHash)
     })
 
+    it('should emit tx:dropped with the txMetas id only after the second call', function (done) {
+    txMeta = {
+      id: 1,
+      hash: '0x0593ee121b92e10d63150ad08b4b8f9c7857d1bd160195ee648fb9a0f8d00eeb',
+      status: 'submitted',
+      txParams: {
+        from: '0x1678a085c290ebd122dc42cba69373b5953b831d',
+        nonce: '0x1',
+        value: '0xfffff',
+      },
+      history: [{}],
+      rawTx: '0xf86c808504a817c800827b0d940c62bb85faa3311a998d3aba8098c1235c564966880de0b6b3a7640000802aa08ff665feb887a25d4099e40e11f0fef93ee9608f404bd3f853dd9e84ed3317a6a02ec9d3d1d6e176d4d2593dd760e74ccac753e6a0ea0d00cc9789d0d7ff1f471d',
+    }
+
+      providerResultStub['eth_getTransactionCount'] = '0x02'
+      providerResultStub['eth_getTransactionByHash'] = {}
+      pendingTxTracker.once('tx:dropped', (id) => {
+        if (id === txMeta.id) {
+          delete providerResultStub['eth_getTransactionCount']
+          delete providerResultStub['eth_getTransactionByHash']
+          return done()
+        } else done(new Error('wrong tx Id'))
+      })
+
+      pendingTxTracker._checkPendingTx(txMeta).then(() => {
+        pendingTxTracker._checkPendingTx(txMeta).catch(done)
```

Comment:
Why are we ignoring the returned promise? Should we be returning it to the chain?

Reply:
test: `should emit tx:dropped with the txMetas id only after the second call`

so we are testing the emit's logic here